### PR TITLE
Update T1580.yaml AWS - EC2 Security Group Enumeration

### PR DESCRIPTION
**Pull request by prashanthpulisetti**
# Simulate Attacker Action for Enumerating EC2 Security Groups

## Details
This update introduces an Atomic Test designed to simulate an attacker's actions in enumerating EC2 Security Groups within a compromised AWS environment. The test leverages the AWS CLI to query and list all security groups, revealing details like group IDs, names, and associated rules. This simulation helps in understanding potential reconnaissance steps an attacker might take after gaining initial access to AWS resources.

### Implementation
- The test uses the `aws ec2 describe-security-groups` command.
- Input parameters include the AWS profile (`aws_profile`) and the desired output format (`output_format`).
- This atomic test is compatible with Linux, macOS, Windows, and AWS as an IaaS platform.
- Prerequisites include the installation and configuration of the AWS CLI, with appropriate permissions to access EC2 Security Group information.

## Testing
- The test was executed in a controlled AWS environment using different profiles with varying permission levels to ensure accurate response based on IAM roles.
- Tested on various platforms including Linux, macOS, and Windows to ensure compatibility and correct execution of commands.
- Validation checks were performed to ensure no configuration changes are made in the AWS environment, confirming the test's read-only nature.

## Associated Issues
- There are no known issues associated with this test. However, users should be aware that running this test in a production environment can potentially reveal sensitive security configuration details. It's recommended to execute this in a test or sandbox AWS environment.
- Care should be taken with the AWS CLI profile used, as profiles with extensive permissions might inadvertently expose more information than intended.

### CODE REVIEW

File: atomics/T1580/T1580.yaml [hunk 1]
```diff
@@ -54,3 +54,45 @@ atomic_tests:
       rm -rf stratus*
     name: sh
     elevation_required: false
+    
+- name: AWS - EC2 Security Group Enumeration
+  description: "Simulate an attacker's action to enumerate EC2 Security Groups in a compromised AWS environment."
+  supported_platforms:
+    - linux
+    - macos
+    - windows
```
Comment (cyberbuff) [inline atomics/T1580/T1580.yaml:63]: Hello @prashanthpulisetti  Thanks for your contribution. This atomic should belong only in `iaas:aws` platform and you should be able to execute it any of the platforms(windows, macos and linux). You can remove these platforms here

**Review by cyberbuff — CHANGES_REQUESTED** [review]

File: atomics/T1580/T1580.yaml (hunk 1 above)
Comment (prashanthpulisetti) [reply atomics/T1580/T1580.yaml:63]: Hi @cyberbuff , thanks for your inputs, let me update the file. Thanks!!!

Comment (cyberbuff) [reply atomics/T1580/T1580.yaml]: Thanks for the fix :)

File: atomics/T1580/T1580.yaml [hunk 2]
```diff
@@ -54,3 +54,46 @@ atomic_tests:
...
+        if ! aws sts get-caller-identity --profile #{aws_profile}; then
+          echo "AWS CLI not properly configured. Please configure AWS CLI."
+        fi
+  
+  cleanup_commands:
+    - description: No specific cleanup required as this test only reads information.
+  
+  notes:
+    - Ensure that this test is conducted in a controlled environment to prevent accidental exposure of sensitive information.
+    - The effectiveness of this test relies on the permissions associated with the AWS profile used. It should have read access to EC2 security groups.
```
Comment (cyberbuff) [inline atomics/T1580/T1580.yaml:99]: Hello @prashanthpulisetti We dont allow additional properties to be specified. Can you remove `notes` and `cleanup_commands` here ?

```
	Additional properties are not allowed ('cleanup_commands', 'notes' were unexpected)

```

Comment (prashanthpulisetti) [reply atomics/T1580/T1580.yaml:99]: Hi @cyberbuff , thanks for your inputs, let me modify

**Review by patel-bhavin — APPROVED** [review]

**Comment by patel-bhavin** [comment]
The PR is updated based on @cyberbuff 's comments  ! thank you for the atomic @prashanthpulisetti

**Review by cyberbuff — APPROVED** [review]

**Comment by prashanthpulisetti** [comment]
@patel-bhavin can you confirm if this PR is still open? or can be closed?

**Comment by patel-bhavin** [comment]
Yes, the PR is good! Thanks for contributing this test !

**Pull request merged**